File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)-1
        ans = 0
        while l<r:
            area = min(nums[l], nums[r])* (r-l)
            ans= max(ans , area)
            if(nums[l]>nums[r]):
                r-=1
            elif(nums[l]<nums[r]):
                l+=1
            else:
                l+=1
                r-+1
        return ans

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int l = 0;
        int r = nums.size()-1;
        int ans = 0;
        while(l<r){
            int water = min(nums[l], nums[r]) * (r-l);
            ans = max(ans, water);
            if(nums[l]<nums[r]) l++;
            else if(nums[l]> nums[r]) r--;
            else l++, r--;
        }
        return ans; 
    }
};